pyproject.toml

```diff
@@ -96,6 +97,7 @@ all = [
     "nmslib", "nmslib-metabrainz",
     "torch", "pytorch-lightning",
     "ipywidgets", "plotly", "nbformat",
+    "cupy-cuda12x",
```

Comment:
Let's probably make a shorter `cupy` extra?
@blondered fyi

Reply:
Yes, that should be covered by `extras = ["cupy"]`, isn't it?

```
cupy-cuda12x = {version = "^13.3.0", python = "<3.13", extras = ["cupy"], optional = true}cupy-cuda12x = {version = "^13.3.0", python = "<3.13", extras = ["cupy"], optional = true}
```

I added cupy extras by using `poetry add` with `--extras` option.
https://python-poetry.org/docs/cli/#options-5